app/services/usage_monitoring/process_all_subscription_activities_service.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+module UsageMonitoring
+  class ProcessAllSubscriptionActivitiesService < BaseService
+    Result = BaseResult
+
+    def call
+      # NOTE: If we need to handle different delays per organization, this would be done here.
+      #     This is why it's a dedicated service and not just done in the job
+
+      Organization.with_activity_tracking.select(:id).find_each do |organization|
```

Comment:
I don't see a reason for this "funky" pattern. I'd rather be explicit and fetch with: `.pluck(:id).each` 

There's hardly any reason to use `find_each` here anyway.

Reply:
Yeah, I can remove the `select` it's silly optimisation 👍

Reply:
Alerting is not going to be a "premium integration", so this should just be `Organization.all` in the end.
And indeed, no need to "batch" since Organization is always expected to be small-ish.